### ISSUE TITLE
navbar button group with dropdown using an anchor tag

### ISSUE THREAD

**Issue by CriseX** [twbs/bootstrap#16268]
This glitches display on the button when dropdown is open. You can get around this with some markup gymnastics and it doesn't really work with a dropdown button inside an existing form.

Broken: http://www.bootply.com/7vhKmOy2sn

Working: http://www.bootply.com/4VrKDN6fz9

The working example though, can not work inside an actual form. Coincidentally, this also presents an additional solution to #16266, but for that this is even more excess markup. I came across both issues while trying to find the leanest possible markup to do something fairly similar to this. The 30 second solution for the dropdown issue is to obviously use jquery event handlers, but it isn't very lean either.

**Comment by twbs-lmvtfy** [comment]
Hi @CriseX!

You appear to have posted a live example (http://s.bootply.com/render/7vhKmOy2sn), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 61, column 21: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 9, column 9: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.3.2
* line 78, column 9: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.3.2

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by twbs-lmvtfy** [comment]
Hi @CriseX!

You appear to have posted a live example (http://s.bootply.com/render/4VrKDN6fz9), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 62, column 21: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 9, column 9: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.3.2
* line 80, column 9: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.4 ; saw what appears to be usage of Bootstrap 3.3.2

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by CriseX** [comment]
Bots, sigh, would have if I could have :).

**Comment by CriseX** [comment]
With a little more fiddling, removing the navbar link list, and having just the button group the glitch seems to go away, so it originates from the set of navbar links. not the dropdown component per se. But the particular mix of the two.

**Comment by cvrebert** [comment]
@CriseX Please use JS Bin instead of Bootply so that you can use Bootstrap v3.3.4. Thanks.

**Comment by CriseX** [comment]
Sure, I can do that, but it shouldn't change anything... I came across the issue 3.3.4, for the record the issue itself looks even more broken with the non-flat theme file applied to it.

Ironically codeply offers the latest version whereas bootply does not, so if you click run on codeply and change the version it will show you the render using 3.3.4, can't save it without creating an account though.

**Comment by cvrebert** [comment]
The navbar isn't terribly flexible (and is going to be overhauled in v4). You can't put arbitrary stuff inside `.navbar-nav`s.
Moving the btn-group outside of the navbar-nav seems to work: http://jsbin.com/yexuwi/2/edit

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (http://jsbin.com/yexuwi/2/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by CriseX** [comment]
You are right it does work, like I said I came across this trying to combine static text label and dropdown button group (the referenced issue) which is why this is one of the approaches I tried (ie, replace the first list item link in the right nav with the desired text).

Looks like this can be closed then.

**Comment by CriseX** [comment]
Actually going to re-open this... see what happens when navbar is collapsed when using your bin http://jsbin.com/yexuwi/4, I see some overflow issues with the navbar and an opened dropdown.

You may close this again once you decide if  this is not something that can be addressed in v3,

**Comment by cvrebert** [comment]
Yeah, it overflows, but you can scroll to reach the rest of it.
Closing as out of scope for v3. Hopefully v4's navbar will be less wonky.
